src/gui/systray.cpp

```diff
@@ -133,123 +126,225 @@ void Systray::setToolTip(const QString &tip)
-    if (QGuiApplication::screens().count() > 1) {
-        trayScreen = QGuiApplication::screens().at(screenIndex());
+void Systray::pauseResumeSync()
+{
+    if (_syncIsPaused) {
+        _syncIsPaused = false;
+        emit resumeSync();
     } else {
-        trayScreen = QGuiApplication::primaryScreen();
+        _syncIsPaused = true;
+        emit pauseSync();
     }
+}
 
-    int screenWidth = trayScreen->geometry().width();
-    int screenHeight = trayScreen->geometry().height();
-    int availableWidth = trayScreen->availableGeometry().width();
-    int availableHeight = trayScreen->availableGeometry().height();
-
-    QPoint topRightDpiAware = QPoint();
-    QPoint topLeftDpiAware = QPoint();
-    if (this->geometry().left() == 0 || this->geometry().top() == 0) {
-        // tray geometry is invalid - QT bug on some linux desktop environments
-        // Use mouse position instead. Cringy, but should work for now
-        topRightDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-    } else {
-        topRightDpiAware = this->geometry().topRight() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = this->geometry().topLeft() / trayScreen->devicePixelRatio();
+/********************************************************************************************/
+/* Helper functions for cross-platform tray icon position and taskbar orientation detection */
+/********************************************************************************************/
+
+QScreen *Systray::currentScreen() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto cursorPos = QCursor::pos();
+    const auto it = std::find_if(screens.cbegin(), screens.cend(), [=](QScreen *screen) {
+        return screen->geometry().contains(cursorPos);
+    });
+    return (it != screens.cend()) ? *it : nullptr;
+}
+
+int Systray::currentScreenIndex() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto screenIndex = screens.indexOf(currentScreen());
+    return screenIndex > 0 ? screenIndex : 0;
+}
+
+Systray::TaskBarPosition Systray::taskbarOrientation() const
+{
+// macOS: Always on top
+#if defined(Q_OS_MACOS)
+    return TaskBarPosition::Top;
+// Windows: Check registry for actual taskbar orientation
+#elif defined(Q_OS_WIN)
+    auto taskbarGeometry = Utility::registryGetKeyValue(HKEY_CURRENT_USER,
+        "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StuckRects3",
+        "Settings");
+    switch (taskbarGeometry.toInt()) {
+    // Mapping windows binary value (0 = left, 1 = top, 2 = right, 3 = bottom) to qml logic (0 = bottom, 1 = left...)
+    case 0:
+        return TaskBarPosition::Left;
+    case 1:
+        return TaskBarPosition::Top;
+    case 2:
+        return TaskBarPosition::Right;
+    case 3:
+        return TaskBarPosition::Bottom;
+    default:
+        return TaskBarPosition::Bottom;
```

Comment:
`case 3` and `default` could be merged, but I guess if you merge it one way you have to add fallthrough attributes, and if you merge it another way you can get warnings that some case is not handled explicitly, so probably this is the best choice here.

Reply:
Yep, I'd keep it like this.